### DISCUSSION TITLE
Is there any feature to move a definition from header to corresponding source file or vice versa?

### DISCUSSION THREAD

**Discussion by starball5** [microsoft/vscode-cpptools#11188] [Q&A] [answered]
Rationale: One can cut and paste, but it's not always that simple. Ex. class member functions need to add `class_name::`, which can be tedious when there are many member functions

If there's currently no such feature, I'll consider raising a feature request.

This was raised [as a question on Stack Overflow](https://stackoverflow.com/q/76570623/11107541) somewhat recently.

**Comment by sean-mcmanus** [comment, marked as answer]
VS 2022 has this feature (Move Definition):

![image](https://github.com/microsoft/vscode-cpptools/assets/19859882/ba8571ef-d418-44dc-8d20-2a6550409935)

We're currently working on the Extract Function refactoring instead https://github.com/microsoft/vscode-cpptools/issues/1162 , but Move Definition could potential be added to VS Code in the future (the work on Extract Function could make the Move Definition work faster to do).
